### PR TITLE
Refactor collection list and editor

### PR DESCRIPTION
**Pull request by rjmackay**
- Move collection toggles alongside other collections code
- Refactor collection editor to use ModalService
- Refactor collection listing to use ModalService
- Fix bugs with collection editor showing wrong collection (ushahidi/platform#1407)
- Update modal container to accept a plain object, not just scopes

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/411)

### CODE REVIEW

File: app/common/directives/modal-container.directive.js [hunk 1]
```diff
@@ -35,12 +35,26 @@ function ModalContainer($timeout, $rootScope, $compile, ModalService) {
...
+        function makeTemplateScope(scope) {
+            // If no scope was passed, create a child of our directive scope
+            if (!scope) {
+                // @todo should this just be $rootScope.$new() ??
+                return $scope.$new();
+            }
+
+            // If scope isn't actually a scope yet, make it into one
+            if (scope.constructor !== $rootScope.constructor) {
+                return angular.extend($rootScope.$new(), scope);
```
Comment (rjmackay) [inline app/common/directives/modal-container.directive.js]: @willdoran I'm particularly pleased how this worked out. Take all the load of managing scope off callers to ModalService.. they can now just pass plain JS objects.
Might be worth doing the same for sliders too..

Only :( is the lack of tests

Comment (willdoran) [inline app/common/directives/modal-container.directive.js]: Yeah it looks very clean, it really pulls the work inside the service :)

**Comment by rjmackay** [comment]
Test on http://qa.fix-collection-editing-14-7.ushahidilab.com/

**Comment by jshorland** [comment]
See notes in #1407

**Pull request merged**